src/rgw/rgw_rest_role.h

```diff
@@ -0,0 +1,116 @@
+#ifndef CEPH_RGW_REST_ROLE_H
+#define CEPH_RGW_REST_ROLE_H
+
+class RGWRestRole : public RGWOp {
+protected:
+  string role_name;
+  string role_path;
+  string trust_policy;
+  string policy_name;
+  string perm_policy;
+  string path_prefix;
+
+public:
+  RGWRestRole() = default;
+  virtual void init(RGWRados *store, struct req_state *s, RGWHandler *h) {
+    RGWOp::init(store, s, h);
+  }
```

Comment:
just saying that these overrides don't do anything other than forward the call to `RGWOp`. if you remove them from `RGWRestRole`, you'll get the same behavior